File: web/src/api/system.ts
import { api } from "./client";

export interface GpuInfo {
  available: boolean;
  used_mb: number;
  total_mb: number;
  name: string | null;
}

export interface HealthResponse {
  status: "ok" | "degraded" | "down";
  db: boolean;
  gpu: GpuInfo;
}

export interface ModelsResponse {
  asr: string[];
  tts: string[];
  cloning: string[];
  separator: string[];
  translation: string[];
}

export const getHealth = () =>
  api.get<HealthResponse>("/health").then((r) => r.data);

export const getModels = () =>
  api.get<ModelsResponse>("/models").then((r) => r.data);

File: src/voxcraft/api/schemas/system.py
"""系统 / 运维 schemas。"""
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


class GpuInfo(BaseModel):
    available: bool = False
    used_mb: int = 0
    total_mb: int = 0
    name: str | None = None


class HealthResponse(BaseModel):
    status: Literal["ok", "degraded", "down"] = "ok"
    db: bool = True
    gpu: GpuInfo = GpuInfo()


class ModelsResponse(BaseModel):
    asr: list[str] = []
    tts: list[str] = []
    cloning: list[str] = []
    separator: list[str] = []
    translation: list[str] = []
